### PR TITLE
33 binary put post support in http request

### PR DESCRIPTION
**Pull request by jowerner**

### CODE REVIEW

File: src/main/java/com/xceptance/xlt/engine/httprequest/HttpRequest.java [hunk 1]
```diff
@@ -445,14 +457,71 @@ public HttpRequest removeParams()
...
+     *            the file from which to read the request body
+     * @throws IOException
+     *             if the file could not be read
+     */
+    public HttpRequest body(final File file) throws IOException
+    {
+        final byte[] bytes = FileUtils.readFileToByteArray(file);
+        body(bytes);
+
+        return this;
```
Comment (h-arlt) [inline src/main/java/com/xceptance/xlt/engine/httprequest/HttpRequest.java:506]: Can be shortened to `return body(bytes);` but doesn't have to.

File: src/main/java/com/xceptance/xlt/engine/httprequest/HttpRequest.java [hunk 2]
```diff
@@ -445,14 +457,71 @@ public HttpRequest removeParams()
...
+     *            the input stream from which to read the request body
+     * @throws IOException
+     *             if the input stream could not be read
+     */
+    public HttpRequest body(final InputStream inputStream) throws IOException
+    {
+        final byte[] bytes = IOUtils.toByteArray(inputStream);
+        body(bytes);
 
         return this;
```
Comment (h-arlt) [inline src/main/java/com/xceptance/xlt/engine/httprequest/HttpRequest.java:526]: Can be shortened to `return body(bytes);` but doesn't have to.

**Review by h-arlt — APPROVED** [review]
Well done.

File: src/main/java/com/xceptance/xlt/engine/httprequest/HttpRequest.java (hunk 2 above)
Comment (jowerner) [reply src/main/java/com/xceptance/xlt/engine/httprequest/HttpRequest.java]: Done.

File: src/main/java/com/xceptance/xlt/engine/httprequest/HttpRequest.java (hunk 1 above)
Comment (jowerner) [reply src/main/java/com/xceptance/xlt/engine/httprequest/HttpRequest.java]: Done.

**Pull request merged**